### PR TITLE
Add composite mode flag to AndroidView

### LINKED ISSUE #55218: Expose Android hybrid views to the framework
`AndroidView` currently renders to a texture. Hybrid composition requires a mechanism for indicating the composite mode.

Due to limitations in the Android SDK [1] that makes it harder to keep this bit as pure implementation detail, a `composeMode` argument should be added to `AndroidView`.  

The argument will default to `texture`.  Therefore, a plugin can override it, and set it as `hybrid`.

[1] Can't submit a single rendered frame, which is the union of Android rendering + Flutter render.

### PR DESCRIPTION
**Pull request by blasten**
## Description

Wire up `compositeMode` argument in `Android View`.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/55218

## Tests

I added the following tests: Initial unit tests for hybrid mode. Additional unit tests will be added as this code path becomes more stable, and it's more clear how to split the state machines for hybrid and texture.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/rendering/platform_view.dart [hunk 1]
```diff
@@ -32,6 +32,17 @@ enum PlatformViewHitTestBehavior {
   transparent,
 }
 
+/// How an Android view is composed along side with Flutter UI.
+enum AndroidViewCompositeMode {
```
Comment (amirh) [inline packages/flutter/lib/src/rendering/platform_view.dart:36]: What do you think about calling this "SurfaceType" and have the values be texture/platformViewLayer ? as this is what the framework knows about (e.g may end up using a different approach on to composite platform view layers one day)

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 2]
```diff
@@ -107,6 +109,9 @@ class AndroidView extends StatefulWidget {
   /// {@endtemplate}
   final TextDirection layoutDirection;
 
+  /// How to compose the Android view.
+  final AndroidViewCompositeMode compositeMode;
```
Comment (amirh) [inline packages/flutter/lib/src/widgets/platform_view.dart:113]: What happens if the widget is being rebuilt with a different surface type?

File: packages/flutter/lib/src/rendering/platform_view.dart (hunk 1 above)
Comment (amirh) [inline packages/flutter/lib/src/rendering/platform_view.dart:36]: What do you think about calling this "SurfaceType" and have the values be texture/platformViewLayer ? as this is what the framework knows about (e.g may end up using a different approach on to composite platform view layers one day)

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 2 above)
Comment (blasten) [reply packages/flutter/lib/src/widgets/platform_view.dart:113]: Assuming we don't leak state in the external view embedder, or in the Android embedding, that would be the equivalent of detaching the previous view, and attaching a new one.

File: packages/flutter/lib/src/rendering/platform_view.dart (hunk 1 above)
Comment (blasten) [reply packages/flutter/lib/src/rendering/platform_view.dart:36]: sgtm

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (blasten) [reply packages/flutter/lib/src/widgets/platform_view.dart:113]: In that case, do you suggest `AndroidViewController` should implement `PlatformViewController`, so an instance is returned from `PlatformViewLink`'s `onCreatePlatformView()`?    Your proposal sounds good to me.

File: packages/flutter/lib/src/rendering/platform_view.dart (hunk 1 above)
Comment (blasten) [reply packages/flutter/lib/src/rendering/platform_view.dart:36]: This depends on https://github.com/flutter/flutter/pull/55175#discussion_r412308174

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (blasten) [reply packages/flutter/lib/src/widgets/platform_view.dart:113]: I wonder if with this approach, the `PlatformViewController` interface would need additional surface that only makes sense on Android.  Currently, this interface is only used for web.

**Comment by goderbauer** [comment]
(PR triage): gentle ping. Is this PR still being worked on?

**Comment by blasten** [comment]
@goderbauer Yep. It's still being worked on. I added the `work-in-progress` label back. Would that be sufficient to hide it from the triage list?

**Comment by blasten** [comment]
Closing this PR in favor of @bparrishMines PR.

**Pull request closed without merging**